### PR TITLE
Spinner redesign in MainActivity

### LINKED ISSUE #562: Redesign the spinner on the MainActivity
Redesign it to make it easier to find chapters in the huge list.

Replace the Spinner with a Button with a right carret that looks like a Spinner. 
On click of the button, re-use the implementation found in Settings.

### PR DESCRIPTION
**Pull request by tasomaniac**
Duplicated the behavior of `LocationListPreference` and created `ChapterSelectDialog`. 

Spinner and its adapter in `MainActivity` is removed. 
Instead we have now just a list of Chapters and a TextView. Clicking on it opens the Dialog with search. 

In xml, I just used `TextView` with `Spinner` style we had before. That turned out pretty good. We have the same look and feel. 

Right now, it still looks like a spinner. This can also be changed. I didn't want to spend too much time on design decisions. 

### Screenshots

Before | After
------ | ------
before| after

Fixes #562

### CODE REVIEW

File: app/src/main/java/org/gdg/frisbee/android/chapter/MainActivity.java [hunk 1]
```diff
@@ -444,6 +428,11 @@ void launchAppInviteActivity(Intent intent) {
         startActivity(newIntent);
     }
 
+    @Override
+    public void onChapterSelected(Chapter selectedChapter) {
+        updateSelectionFor(selectedChapter.getGplusId());
```
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/chapter/MainActivity.java]: here, you convert a chapter into an id and then back into a chapter. Should be made more performant.

File: app/src/main/java/org/gdg/frisbee/android/chapter/MainActivity.java [hunk 2]
```diff
@@ -327,7 +323,7 @@ protected String getTrackedViewName() {
         } catch (IndexOutOfBoundsException e) {
             pageName = "";
         }
-        Chapter chapter = mChapterAdapter.findById(selectedChapterId);
+        Chapter chapter = findChapterById(selectedChapterId);
```
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/chapter/MainActivity.java]: this could use chapter as a variable instead of searching for the chapter for each track event (not part of this PR though)

File: app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        View view = getActivity().getLayoutInflater()
+            .inflate(R.layout.view_location_list_preference, (ViewGroup) getView(), false);
+        ButterKnife.bind(this, view);
+
+        return new AlertDialog.Builder(getActivity())
+            .setNegativeButton(R.string.feedback_cancel, null)
```
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java]: feedback_cancel?

File: app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        args.putParcelable(EXTRA_SELECTED_CHAPTER, selectedChapter);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        chapters = getArguments().getParcelableArrayList(EXTRA_CHAPTERS);
```
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java]: maybe it is better to get the chapters out of the cache instead of parceling in and out.

File: app/src/main/java/org/gdg/frisbee/android/chapter/MainActivity.java (hunk 2 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/chapter/MainActivity.java]: I think as a field we should also store `selectedChapter` as a whole object instead of just the id. I wanted to do it but I think it will blow up this PR.

File: app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java (hunk 3 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java]: I was just being lazy here. `feedback_cancel` is just cancel. That is something we add as a string for the feedback dialog. 
I think we should change this string to just `cancel`

File: app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java (hunk 4 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java]: Then, do I have to add the logic to get them from network if they are not available in the cache? Is that even be possible? 
If we are going to use it in Settings, it will definitely be possible.

**Comment by tasomaniac** [comment]
@friedger Screen recordings are added

File: app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java (hunk 3 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java]: I just used `android.R.string.cancel` now 👍

**Comment by friedger** [comment]
It is available in the cache as it is now used in the extra. For the edge
case where the cache is cleared due to low resources we can except the
extra delay.

Le 15 sept. 2016 00:28, "Said Tahsin Dane" <notifications@github.com> a
écrit :

**Comment by tasomaniac** [comment]
I was just hesitant to add network calls into the dialog. It was because the dialog itself is so simple now. Let me look at it.

**Comment by friedger** [comment]
Maybe not now.
But querying a repository should be easy ☺️

Le 15 sept. 2016 09:16, "Said Tahsin Dane" <notifications@github.com> a
écrit :

**Comment by tasomaniac** [comment]
Most comments are addressed. 
No network call currently done in the dialog. It's just use the cache. It has a TODO in it as well. I think it will be fine. 
The other thing left is to adjust DialogPreference to use this Dialog. I want to do it in a separate PR. The problem is that it is not easy to use DialogFragment inside DialogPreference. I think the new preference support lib does it. I want to check that to see if it is easier with that.

**Comment by friedger** [comment]
:+1:
